go-controller/pkg/ovn/policy_test.go

```diff
@@ -1655,6 +1655,49 @@ var _ = ginkgo.Describe("OVN NetworkPolicy Operations", func() {
 			err := app.Run([]string{app.Name})
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 		})
+
+		ginkgo.It("can handle network policy with equivalent rules", func() {
+			app.Action = func(ctx *cli.Context) error {
+				namespace1 := *newNamespace(namespaceName1)
+				peer := knet.NetworkPolicyPeer{
+					IPBlock: &knet.IPBlock{
+						CIDR: "1.1.1.1",
+					},
+				}
+				// equivalent rules in one peer
+				networkPolicy1 := newNetworkPolicy(netPolicyName1, namespace1.Name, metav1.LabelSelector{},
+					[]knet.NetworkPolicyIngressRule{{
+						From: []knet.NetworkPolicyPeer{peer, peer},
+					}}, nil)
+				// equivalent rules in different peers
+				networkPolicy2 := newNetworkPolicy(netPolicyName2, namespace1.Name, metav1.LabelSelector{},
+					[]knet.NetworkPolicyIngressRule{
+						{
+							From: []knet.NetworkPolicyPeer{peer},
+						},
+						{
+							From: []knet.NetworkPolicyPeer{peer},
+						},
+					}, nil)
+				startOvn(initialDB, []v1.Namespace{namespace1}, []knet.NetworkPolicy{*networkPolicy1, *networkPolicy2},
+					nil, nil)
+				expectedData := initialDB.NBData
+				policy1DB := getPolicyData(networkPolicy1, nil, []string{}, nil, "",
+					false, false)
+				expectedData = append(expectedData, policy1DB...)
+				policy2DB := getPolicyData(networkPolicy2, nil, []string{}, nil, "",
+					false, false)
+				expectedData = append(expectedData, policy2DB...)
+				defaultDenyDb := getDefaultDenyData(networkPolicy1, nil, "", false)
+				expectedData = append(expectedData, defaultDenyDb...)
```

Comment:
nit: maybe also move all this into an helper ? 

```golang
func expectedDataForNetworkPolicies(initialDBState []ACL, netPolicy ...NetworkPolicy) []ACL {
...
}
```

It would allow your test to look like:
```golang
gomega.Eventually(fakeOvn.nbClient).Should(
    libovsdb.HaveData(
        expectedDataForNetworkPolicies(
            initialDB.NBData,
            networkPolicy1,
            networkPolicy2,
        )
    )
)
```

Reply:
it has a bit more args to cover all test cases, but the idea is the same